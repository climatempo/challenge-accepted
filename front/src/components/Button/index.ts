import styled from 'styled-components';

const Button = styled.button`
  background: none;
  border: none;
  cursor: pointer;
  padding: 0;
  height: fit-content;
  width: fit-content;
`;

export default Button;
