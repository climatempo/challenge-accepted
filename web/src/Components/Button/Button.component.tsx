import { HTMLAttributes } from 'react';
import { IconButton } from '@mui/material';

function Button(props: HTMLAttributes<HTMLButtonElement>) {
    const { children, onClick, title } = props;

    return <IconButton className="h-14 w-14"
                       onClick={ onClick }
                       title={ title }>
        { children }
    </IconButton>;
}

export default Button;