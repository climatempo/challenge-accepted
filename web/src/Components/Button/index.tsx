import Button from './Button.component';

export default Button;