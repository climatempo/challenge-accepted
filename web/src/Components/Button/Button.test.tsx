import renderer from 'react-test-renderer';
import Button from './index';

test('Button', () => {
    const button = renderer.create(
      <Button title="This is a test title" onClick={() => "This is a test function"}>
          This is a test children
      </Button>
    );

    expect(button.toJSON()).toMatchSnapshot();
});