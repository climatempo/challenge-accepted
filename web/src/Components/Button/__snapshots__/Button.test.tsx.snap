// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium h-14 w-14 css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="This is a test title"
  type="button"
>
  This is a test children
</button>
`;
